code/modules/power/smes.dm

```diff
@@ -206,156 +206,83 @@
-		height: 8px;
-		max-height: 8px;
-		background: #363;
-		border: 1px solid white;
-		padding: 1px;
-		position: relative;
-		margin: 0.25em 0;
-	}
-
-	.bar .inner {
-		height: 100%;
-		margin: 0;
-		padding: 0;
-		background: #4e4;
-	}
-
-	.bar .marker {
-		position: absolute;
-		top: 0px;
-		width: 1px;
-		margin-left: -1px;
-		height: 40%;
-		background: white;
-		border: 1px solid black;
-		border-top: none;
-	}
-
-	p {
-		margin: 0.25em 0;
-	}
-</style>
-<div id="#powerMenu">
-<h3 style="text-align: center;">SMES Power Storage Unit [n_tag ? "- [n_tag]" : null]</h3>
-<br>
-	<strong>Stored Charge: [round(100.0*charge/capacity, 0.1)]%</strong> <em>([charging ? "Charging" : ((chargecount > 0) ? "Preparing to charge..." : "Not Charging")])</em>
-	<div class='bar'><div class='inner' style="width: [round(100 * (charge / capacity), 0.01)]%;"></div></div>
-<br>
-	<strong>Charging:</strong> [chargemode ? "<b>Enabled</b> (<a href='?src=\ref[src];cmode=1'>Disable</a>)" : "<b>Disabled</b> (<a href='?src=\ref[src];cmode=1'>Enable</a>)"]
-	<br><strong>Input level:</strong> <a href='?src=\ref[src];input=set'>[chargelevel]</a> (<a href='?src=\ref[src];input=min'>Min</a> &middot; <a href='?src=\ref[src];input=max'>Max</a>)
-	<div class='bar-outer'><div class='bar'><div class='inner' style="width: [max(0, min(100, round(100 * (lastexcess / SMESMAXCHARGELEVEL), 0.01)))]%;"></div><div class='marker' style="left: [round(100 * (chargelevel / SMESMAXCHARGELEVEL), 0.01)]%;"></div></div>
-	<strong>Available:</strong> [round(lastexcess)] W
-<br>
-<br>
-	<strong>Output:</strong> [online ? "<b>Enabled</b> (<a href='?src=\ref[src];online=1'>Disable</a>)" : "<b>Disabled</b> (<a href='?src=\ref[src];online=1'>Enable</a>)"]
-	<br><strong>Output level:</strong> <a href='?src=\ref[src];output=set'>[output]</a> (<a href='?src=\ref[src];output=min'>Min</a> &middot; <a href='?src=\ref[src];output=max'>Max</a>)
-	<div class='bar'><div class='inner' style="width: [min(100, round(100 * (loaddemand / SMESMAXOUTPUT), 0.01))]%;"></div><div class='marker' style="left: [round(100 * (output / SMESMAXCHARGELEVEL), 0.01)]%;"></div></div>
-	<strong>Current load:</strong> [round(loaddemand)] W
-</div>
-"}
-
-	user.Browse(t, "window=smes;size=400x340")
-	onclose(user, "smes")
-	return
-
-/obj/machinery/power/smes/Topic(href, href_list)
-	..()
-
-	if (usr.stat || usr.restrained() )
+/obj/machinery/power/smes/ui_state(mob/user)
+	return tgui_default_state
+
+/obj/machinery/power/smes/ui_status(mob/user, datum/ui_state/state)
+	return min(
+		state.can_use_topic(src, user),
+		tgui_broken_state.can_use_topic(src, user),
+		tgui_not_incapacitated_state.can_use_topic(src, user)
+	)
+
+/obj/machinery/power/smes/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Smes", src.name)
+		ui.open()
+
+/obj/machinery/power/smes/ui_data(mob/user)
+	var/list/data = list()
+	data["capacityPercent"] = (src.charge / src.capacity * 100)
```

Comment:
Personal take: I would prefer `data["capacity"] = src.capacity` is sent instead of this then this calculation done by the UI. It's super-minor, NBD if you leave it.